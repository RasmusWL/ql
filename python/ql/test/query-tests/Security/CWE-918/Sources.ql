import python
import TaintSetup

from HttpRequestTaintSource source, TaintKind kind, AstNode ast
where
    source.isSourceOf(kind) and
    ast = source.(ControlFlowNode).getNode()
select ast, ast.getScope(), kind
