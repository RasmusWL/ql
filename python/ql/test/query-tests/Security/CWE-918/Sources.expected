| test.py:20:12:20:18 | request | test.py:20:1:20:20 | Function vuln_1 | django.request.HttpRequest |
| test.py:26:12:26:18 | request | test.py:26:1:26:20 | Function vuln_2 | django.request.HttpRequest |
| test.py:35:12:35:18 | request | test.py:35:1:35:20 | Function vuln_3 | django.request.HttpRequest |
| test.py:44:12:44:18 | request | test.py:44:1:44:20 | Function vuln_4 | django.request.HttpRequest |
| test.py:53:20:53:26 | request | test.py:53:1:53:28 | Function bad_sanitizer1 | django.request.HttpRequest |
| test.py:66:20:66:26 | request | test.py:66:1:66:28 | Function bad_sanitizer2 | django.request.HttpRequest |
| test.py:80:20:80:26 | request | test.py:80:1:80:28 | Function bad_sanitizer3 | django.request.HttpRequest |
| test.py:110:12:110:18 | request | test.py:110:1:110:20 | Function safe_1 | django.request.HttpRequest |
| test.py:121:12:121:18 | request | test.py:121:1:121:20 | Function safe_2 | django.request.HttpRequest |
| test.py:138:12:138:18 | request | test.py:138:1:138:20 | Function safe_3 | django.request.HttpRequest |
| test.py:157:12:157:18 | request | test.py:157:1:157:20 | Function safe_4 | django.request.HttpRequest |
